plugins/catalog-backend/migrations/20221109192547_search_add_original_value_column.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @param { import("knex").Knex } knex
+ */
+exports.up = async function up(knex) {
+  await knex.schema.alterTable('search', table => {
+    table
+      .string('original_value')
+      .nullable()
+      .comment('Holds the corresponding original case sensitive value');
+    table.index(['original_value'], 'search_original_value_idx');
+  });
+};
+
+/**
+ * @param { import("knex").Knex } knex
+ */
+exports.down = async function down(knex) {
+  await knex.schema.alterTable('search', table => {
+    table.dropIndex([], 'search_entity_id_idx');
+    table.dropIndex([], 'search_key_idx');
+    table.dropIndex([], 'search_original_value_idx');
+    table.dropIndex([], 'search_value_idx');
+  });
+  await knex.schema.dropTable('search');
```

Comment:
Oh no, this teardown is way too ambitious :) Do not drop the table. Do not delete unrelated indices; drop the column instead.